osu.Game/Graphics/Gameplay/Catch/Catcher.cs

```diff
@@ -0,0 +1,226 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Linq;
+using OpenTK;
+using OpenTK.Graphics;
+using OpenTK.Input;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
+using osu.Framework.Input;
+using osu.Framework.Timing;
+
+namespace osu.Game.Graphics.Gameplay.Catch
+{
+    public class Catcher : Container

```

Comment:
I'd probably call this class a CatcherController or CatcherContainer, and call the actual Catcher sprite Catcher.

This would also mean moving leaving the general handling logic (bounds checks, input hadnling etc.) in the controller class but having all animation/state changes for the catcher inside the Catcher class.